service_client.go

```diff
@@ -114,6 +114,8 @@ func (client *ServiceClient) setMicroversionHeader(opts *RequestOpts) {
 		opts.MoreHeaders["X-OpenStack-Nova-API-Version"] = client.Microversion
 	case "sharev2":
 		opts.MoreHeaders["X-OpenStack-Manila-API-Version"] = client.Microversion
+	case "volume":
+		opts.MoreHeaders["X-OpenStack-Volume-API-Version"] = client.Microversion
```

Comment:
The `client.Microversion` is not set anywhere currently so it's an empty string. Shouldn't CinderV3 API reject packets that contain an empty string in the `X-OpenStack-Volume-API-Version` field.